### PR TITLE
Port forward schema gen

### PR DESCRIPTION
**Pull request by SimonRichardson**
## Description of change

This is a port forward of #10336 - with the **addition** of the required
`make rebuild-schema`.

## QA steps

Tests pass

### CODE REVIEW

**Comment by SimonRichardson** [comment]
Failed because of windows tar file.

!!build!!

File: Gopkg.toml [hunk 1]
```diff
@@ -535,3 +535,7 @@
 [[constraint]]
   name = "k8s.io/apiextensions-apiserver"
   revision = "4d05e43ad98c1edcf2f52291685bd9bbc12fd2cd"
+
+[[constraint]]
+  revision = "66506e52db051b5e238aa63ec4dad0a790fbc236"
+  name = "github.com/bcsaller/jsonschema"
```
Comment (jameinel) [inline Gopkg.toml]: newline at end of file?

Comment (SimonRichardson) [reply Gopkg.toml]: damn it, i'm installing a plugin into my IDE....

**Comment by SimonRichardson** [comment]
$$merge$$

**Pull request merged**